### ISSUE TITLE
Support for `Float32` / `Float64` Images

### ISSUE THREAD

**Issue by RoyiAvital** [dfchai/Rooted-Spanning-Superpixels#1]
Many images are in the range `[0, 1]` and represented by either `Float32` and `Float64`.

I was wondering what should be changed in order to make the code support those?
Looking at the code I can see mostly memory copy and comparisons which can be easily adapted. Are there more?

**Comment by dfchai** [comment]
As described in the paper, floats cost need to be quantized into integers before using the code.

**Comment by RoyiAvital** [comment]
Is the quantization of 8 bit only hurts performance?

**Comment by dfchai** [comment]
No, no restriction on the number of bits and the number of integers. 
By the way, you can also compare float costs directly. Comparing floats is more accurate than comparing integers. But you need to revised the code to compare floats and need to maintain a queue of float costs of the candidates. Therefore, the speed will be much slower.

**Comment by RoyiAvital** [comment]
The question is, given an image in `[0, 1]` what would be the best way to handle it?
The current code requires to quantize it into Unsigned Char (8 Bits).
Wouldn't that hurt the quality of the Super Pixels?

**Comment by dfchai** [comment]
The best way is to handle the costs as integers. 
If you need more integers, you can easily modify it to accept 16 bits numbers. 
This quantization has little influence on segmentation, which is supported by the experiments described in the paper.

**Comment by RoyiAvital** [comment]
It would be great to add support for 16 and 32 bit quantization (`uint16`, `uint32`).
